server/src/main/java/org/apache/cloudstack/vm/UnmanagedVMsManagerImpl.java

```diff
@@ -551,7 +551,7 @@ private StoragePool getStoragePool(final UnmanagedInstanceTO.Disk disk, final Da
             List<StoragePoolVO> pools = primaryDataStoreDao.listPoolsByCluster(cluster.getId());
             pools.addAll(primaryDataStoreDao.listByDataCenterId(zone.getId()));
             for (StoragePool pool : pools) {
-                if (pool.getPath().endsWith(dsName)) {
+                if (StringUtils.containsAny(pool.getPath(), dsName, dsPath)) {
```

Comment:
aren't we replacing a bug with a bug here (nitpicking maybe but genuine question)
`path.endswith("bla")` is different from `StringUtils.containsAny(path, "bla", "blob")`
considder paths:
- "bla/blah"
- "bla/blob"
- "bla/bleb"
Neither would match the first condition, and though the first is supposed to match the second condition (as I understand) the other two seem false positives to me. 

Am I misinterpreting this @slavkap ?

Reply:
Adding to @DaanHoogland's comment, perhaps we can add a condition to keep the current check for non RBD disks? Raising a concern since the Vmware import of unmanaged instances uses the same code

Reply:
thanks @DaanHoogland, @nvazquez! I've changed it to check only path vs dsPath. Neither for KVM/VMware we need to check for the datastore name in the primary storage path